src/lib/GovernorTimelockCompound.sol

```diff
@@ -0,0 +1,257 @@
+ * {GovernorTimelockCompound-__acceptAdmin} is available to accept ownership of the timelock.
+ *
+ * Using this model means the proposal will be operated by the {TimelockController} and not by the
+ * {Governor}. Thus,
+ * the assets and permissions must be attached to the {TimelockController}. Any asset sent to the
+ * {Governor} will be
+ * inaccessible.
+ *
+ * _Available since v4.3._
+ */
+abstract contract GovernorTimelockCompound is IGovernorTimelock, Governor {
+  using SafeCast for uint256;
+  using Timers for Timers.Timestamp;
+
+  struct ProposalTimelock {
+    Timers.Timestamp timer;
+  }
+
+  // The interface for this variable was changed to conform to the PoolTogether Timelock interface.
+  //
+  // Original openzeppelin:
+  // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L31
+  IPoolTogetherTimelock private _timelock;
+
+  mapping(uint256 => ProposalTimelock) private _proposalTimelocks;
+
+  /**
+   * @dev Emitted when the timelock controller used for proposal execution is modified.
+   */
+  event TimelockChange(address oldTimelock, address newTimelock);
+
+  /**
+   * @dev Set the timelock.
+   */
+  // The timelock interface was changed from the original Openzeppelin source to conform to the
+  // PoolTogether interface.
+  //
+  // Original Openzeppelin source:
+  // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L43
+  constructor(IPoolTogetherTimelock timelockAddress) {
+    _updateTimelock(timelockAddress);
+  }
+
+  /**
+   * @dev See {IERC165-supportsInterface}.
+   */
+  function supportsInterface(bytes4 interfaceId)
+    public
+    view
+    virtual
+    override(IERC165, Governor)
+    returns (bool)
+  {
+    return
+      interfaceId == type(IGovernorTimelock).interfaceId || super.supportsInterface(interfaceId);
+  }
+
+  /**
+   * @dev Overridden version of the {Governor-state} function with added support for the `Queued`
+   * and `Expired` status.
+   */
+  // The _timelock.gracePeriod() was changed from _timelock.GRACE_PERIOD() in the original
+  // Openzeppelin contract.
+  //
+  // Original Openzeppelin source:
+  // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L67
+  function state(uint256 proposalId)
+    public
+    view
+    virtual
+    override(IGovernor, Governor)
+    returns (ProposalState)
+  {
+    ProposalState status = super.state(proposalId);
+
+    if (status != ProposalState.Succeeded) return status;
+
+    uint256 eta = proposalEta(proposalId);
+    if (eta == 0) return status;
+    else if (block.timestamp >= eta + _timelock.gracePeriod()) return ProposalState.Expired;
+    else return ProposalState.Queued;
```

Comment:
nit, diffing this file against the OZ version and we're a bit inconsistent with formatting changes:
- in various places we let `forge fmt` make it's changes, such as flattening these various statements or modifying stlyes of functions sigs
- but we also leave the `/** */` style natspec comments instead of changing to the `///` style we usually use and use in `PoolTogetherGovernor` 

Ideally we should either (1) change comments to match our style, or (2) use the `forgefmt: disable` syntax to prevent auto-formatting this file. I'm ok with either (or with leaving it if we're ok with the current style)

Reply:
https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/20/commits/7445ab9b7d3b04e10af14887db666cd78b0852f5 I went the `forgefmt: disable` router